api/src/main/java/org/openmrs/module/insuranceclaims/api/service/impl/BillServiceImpl.java

```diff
@@ -1,11 +1,61 @@
 package org.openmrs.module.insuranceclaims.api.service.impl;
 
+import org.openmrs.module.insuranceclaims.api.dao.BillDao;
 import org.openmrs.module.insuranceclaims.api.model.Bill;
+import org.openmrs.module.insuranceclaims.api.model.PaymentStatus;
+import org.openmrs.module.insuranceclaims.api.model.ProcessStatus;
+import org.openmrs.module.insuranceclaims.api.model.ProvidedItem;
 import org.openmrs.module.insuranceclaims.api.service.BillService;
+import org.openmrs.module.insuranceclaims.api.service.ProvidedItemService;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
+import java.math.BigDecimal;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+
 @Service("insuranceclaims.BillService")
 @Transactional
 public class BillServiceImpl extends BaseOpenmrsDataService<Bill> implements BillService {
+
+    private static final int DAYS = 7;
+
+    @Autowired
+    private ProvidedItemService providedItemService;
+
+    @Autowired
+    private BillDao billDao;
+
+    @Override
+    public Bill generateBill(List<ProvidedItem> providedItems) {
+
+        Date date = new Date();
```

Comment:
we should do it in separate component

Reply:
verify how it saves in db, I suppose it is in UTC

Reply:
Probably it will be the best to create a separate Util which can be used to work with dates